tests/trinity/integration/test_boot.py

```diff
@@ -0,0 +1,107 @@
+    contains_all
+)
+
+
+# IMPORTANT: Test names are intentionally short here because they end up
+# in the path name of the isolated Trinity paths that pytest produces for
+# us.
+# e.g. /tmp/pytest-of-circleci/pytest-0/popen-gw3/test_light_boot_comman0/xdg/mainnet/jsonrpc.ipc)
+#
+# However, UNIX IPC paths can only be 100 chars which means long paths
+# *WILL* break these tests. See: https://unix.stackexchange.com/q/367008
+
+# This fixture provides a tear down to run after each test that uses it.
+# This ensures the AsyncProcessRunner will never leave a process behind
+@pytest.fixture(scope="function")
+def async_process_runner():
+    runner = AsyncProcessRunner()
+    yield runner
+    runner.kill()
+
+# Great for debugging the AsyncProcessRunner
+# @pytest.mark.asyncio
+# async def test_ping(async_process_runner):
+#     await async_process_runner.run(['ping', 'www.google.de'])
+#     assert await contains_all(async_process_runner.iterate_stdout(), ['byytes from'])
+
+
+@pytest.mark.parametrize(
+    'command',
+    (
+        ['trinity'],
+        ['trinity', '--ropsten'],
+    )
+)
+@pytest.mark.asyncio
+async def test_full_boot(async_process_runner, command):
+    # UPNP discovery can delay things, we use a timeout longer than the discovery timeout
+    await async_process_runner.run(command, timeout_sec=40)
+    assert await contains_all(async_process_runner.iterate_stderr(), {
+        "Started DB server process",
+        "Started networking process",
+        "Running server",
+        "IPC started at",
+    })
+
+
+@pytest.mark.parametrize(
+    'command',
+    (
+        ['trinity', '--tx-pool'],
+        ['trinity', '--tx-pool', '--ropsten'],
+    )
+)
+@pytest.mark.asyncio
+async def test_txpool_full_boot(async_process_runner, command):
+    # UPNP discovery can delay things, we use a timeout longer than the discovery timeout
+    await async_process_runner.run(command, timeout_sec=40)
+    assert await contains_all(async_process_runner.iterate_stderr(), {
+        "Started DB server process",
+        "Started networking process",
+        "Running Tx Pool",
+        "Running server",
+        "IPC started at",
+    })
+
+
+@pytest.mark.parametrize(
+    'command',
+    (
+        ['trinity', '--light'],
+        ['trinity', '--light', '--ropsten'],
+    )
+)
+@pytest.mark.asyncio
+async def test_light_boot(async_process_runner, command):
+    # UPNP discovery can delay things, we use a timeout longer than the discovery timeout
+    await async_process_runner.run(command, timeout_sec=40)
+    assert await contains_all(async_process_runner.iterate_stderr(), {
+        "Started DB server process",
+        "Started networking process",
+        "IPC started at",
+    })
+
+
+@pytest.mark.parametrize(
+    'command',
+    (
+        # mainnet
+        ['trinity'],
+        ['trinity', '--light'],
+        # ropsten
+        ['trinity', '--ropsten'],
+        ['trinity', '--light', '--ropsten'],
```

Comment:
There's a separate test for `--tx-pool` because we also want to check for tx pool related output in that case https://github.com/ethereum/py-evm/pull/1049/files#diff-4efbd71fd82c31c1cbb0ade62cadfa3aR65

Reply:
Yes, but that test doesn't:
- look for exceptions
- run light sync
- run light ropsten sync

Reply:
I'll comment on that later. Mobile s*cks!

Reply:
>run light mainnet sync
>run light ropsten sync

I wasn't exactly sure what we want for the light mode in regard of the transaction pool. Currently, the transaction pool doesn't work for the light mode anyway (see https://github.com/ethereum/py-evm/issues/1031) so I'm inclined to catch that in the tx pool plugin and then add a test that proves that the tx pool **does not start** in light mode (as in, if `--tx-pool` is specified you'll get a warning that tx pool doesn't work yet in light mode).

>look for exceptions

True that. I'll add the `--txpool` setup to the test that scans for exceptions.